src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -10,7 +10,7 @@
 public class Tag {
 
     public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric";
-    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}-]+";
```

Comment:
This will allow the user to start and end with '-' 
Not sure if that was intended.

Reply:
Oh okay, i think I'll change it to only allow for the case where 
1. cannot start or end with `-`
2. allow multiple `-` usecases